Source/WebCore/rendering/style/StyleCachedImage.cpp

```diff
@@ -278,6 +318,9 @@ RefPtr<Image> StyleCachedImage::image(const RenderElement* renderer, const Float
     if (auto renderSVGResource = this->renderSVGResource(renderer))
         return SVGResourceImage::create(*renderSVGResource, reresolvedURL(renderer->document()));
 
+    if (auto renderSVGResource = this->legacyRenderSVGResource(renderer))
```

Comment:
This adds new code for legacy? Why is this necessary?

Reply:
Note, since the aim is to remove legacy, I did not try any fancy solutions like templates and stuff, so things look a bit duplicated, but I think it is okay for now.